src/python/WMCore/MicroService/Unified/MSMonitor.py

```diff
@@ -17,123 +21,170 @@ class MSMonitor(MSCore):
+             "lastupdate": 123,
              "transfers": [rec1, rec2, ... ]
             }
             where each record has the following format:
-            {"dataset":"/a/b/c", "dataType": "primary", "transferIDs": [1,2], "completion":0}
+            {"dataset":"/a/b/c", "dataType": "primary", "transferIDs": [1,2], "completion": 0}
 
-        :param reqName: name of the request
-        :return: list of of status records
-        """
-        transferRecords = []
-
-        # in our case workflow name is the same as request name
-        wname = reqName
-        # get existing transfer IDs record
-        doc = self.getTransferInfo(wname)
-        if not doc:
-            self.logger.error("Failed to find a transfer document for request: %s", wname)
-            return transferRecords
-
-        # loop over all records and obtain new transfer IDs
-        for rec in doc['transfers']:
-            dataset = rec['dataset']
-            dtype = rec['dataType']
-            # obtain new transfer ids for given request name and dataset
-            tids, completion = self.getTransferIds(dataset)
-            rec = {'dataset': dataset, 'dataType': dtype,
-                   'transferIDs': tids, "completion": completion}
-            transferRecords.append(rec)
-        return transferRecords
-
-    def pileupTransfersCompleted(self, transferRecords):
-        "Check if pileup transfers are completed for given transfer records"
-        # FIXME: completion threshold should come either from unified
-        # or campaign configuration, e.g.
-        # thr = self.unifiedConfig().get('completedThreshold', 100)
-        thr = 100
-
-        records = 0
-        transferTypes = ['primary', 'secondary']
-        # loop over all records and obtain new transfer IDs
-        for rec in transferRecords:
-            ctype = rec['dataType']
-            completion = rec['completion']
-            # count records above threshold
-            if ctype in transferTypes and completion > thr:
-                records += 1
-        return True if records == len(transferRecords) else False
-
-    def completion(self, records):
-        """
-        Helper function to calculate completion of given records
-        :param records: list of status records
-        :return: completion value
+
+        :param wname: workflow name
+        :return: a generator of transfer records
         """
-        # may need to implement proper algorithm, so far we take average
-        val = [r['completion'] for r in records]/len(records)
-        return val
+        records = self.reqmgrAux.getTransferInfo(wname)
+        # according to https://github.com/dmwm/WMCore/wiki/ReqMgr2-MicroService-Monitor
+        # we may need to select only record since last updated according
+        # to configuration UpdateInterval setting
+        for rec in records:
+            if self.updateInterval:
+                if time.time()-rec['lastupdate'] > self.updateInterval:
```

Comment:
`lastUpdate`

Reply:
Alan, that's why I wrote about. PLEASE BE CONSISTENT in naming convention through the code. You used `CampaignName`, `dataType` and `lastupdate` (see https://gist.github.com/amaltaro/72599f995b37a6e33566f3c749143154) ALL OF THEM are different naming conventions.

I'll fix it here , but I followed what was in your gist document.

Reply:
Sorry, let me update that gist with the correct key name (camelCase).